inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -137,11 +142,28 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
 
         std::string defaultDeviceID = "";
         DeviceIDParser parsed{deviceName};
-        if (parsed.getDeviceID().empty())
+        std::string deviceid = parsed.getDeviceID();
+        if (deviceid.empty()) {
             defaultDeviceID = getDefaultDeviceID(deviceName);
+            deviceid = defaultDeviceID;
+        }
 
+        std::string fullDeviceName = "";
+        std::string uniqueName = "";
+        std::vector<std::string> supportedMetrics = GetCore()->GetMetric(parsed.getDeviceName(), METRIC_KEY(SUPPORTED_METRICS));
+        if (std::find(supportedMetrics.begin(), supportedMetrics.end(), METRIC_KEY(FULL_DEVICE_NAME)) != supportedMetrics.end()) {
+            fullDeviceName = GetCore()->GetMetric(parsed.getDeviceName(), METRIC_KEY(FULL_DEVICE_NAME)).as<std::string>();
+        }
+        if (fullDeviceName.empty()) {
+            uniqueName = parsed.getDeviceName() + "_" + deviceid;
+        } else {
+            uniqueName = fullDeviceName + "_" + deviceid;
+        }
+
+        // is it right to use deviceNameWithID as deviceName in DeviceInformation?
+        // if GetCore()->GetMetric(deviceNameWithID) is ok ?
         // create meta device
-        metaDevices.push_back({ deviceName, getDeviceConfig(deviceName), numRequests, defaultDeviceID });
+        metaDevices.push_back({ deviceName, getDeviceConfig(deviceName), numRequests, defaultDeviceID, uniqueName});
```

Comment:
hi maxim, when I write test code for ParseMetaDevices, I found it, it maybe a issue, but I am not sure, please to have a look.
is it ok to use deviceNameWithID as deviceName in DeviceInformation?

Reply:
the pure device name like GPU ( whihc translates to the plugin) were separated from the deviceIDs so that user can set different settings for the GPU.0 and GPU.1
https://github.com/openvinotoolkit/openvino/pull/7311

its ok to keep them separately, but for actual loading, configuration, scheduling, etc the fully-qualified device name (name+id) should be used.... from what I see the deviceName in your code is fully-quilified.

we should have tests with GPU.0,GPU.1 etc